trieste/acquisition/function.py

```diff
@@ -790,11 +795,15 @@ def __init__(
 
         if penalizer is None:
             penalizer = soft_local_penalizer
-
         self._lipschitz_penalizer = penalizer
 
+        if base_acquisition_function_builder is None:
+            base_acquisition_function_builder = ExpectedImprovement()
+        self._base_builder = base_acquisition_function_builder
+
         self._lipschitz_constant = None
         self._eta = None
+        self._base_acquisition_function = None
```

Comment:
You'll need a type annotation here or mypy will guess wrongly that the type is always None.

```suggestion
        self._base_acquisition_function : Optional[AcquisitionFunction] = None
```